packages/frontend/src/app/cart/NewCart.tsx

```diff
@@ -7,40 +7,52 @@ import React, { useState, useEffect, useRef } from "react";
 import Image from "next/image";
 import { useStoreContext } from "@/context/StoreContext";
 import Button from "@/app/common/components/Button";
-import { ItemId } from "@/types";
-import { ItemState } from "@/context/types";
+import { ItemId, OrderId, Order } from "@/types";
 import { useRouter } from "next/navigation";
 import { useMyContext } from "@/context/MyContext";
 
-const NewCart = ({ next }: { next: () => void }) => {
-  const { orderItems, products, orderId, updateOrder } = useStoreContext();
-  const [activeCartItems, setActiveCartItems] = useState<ItemState | null>(
-    null,
-  );
+const NewCart = ({
+  next,
+  orderId,
+}: {
+  next: () => void;
+  orderId: OrderId | null;
+}) => {
+  const { stateManager, selectedCurrency } = useStoreContext();
+  const [cartItemIds, setItemIds] = useState<Order["items"] | null>(null);
+  const [cartItemsMap, setCartMap] = useState(new Map());
   const [errorMsg, setErrorMsg] = useState("");
-  const [baseCurrSymbol, setBaseCurrSymbol] = useState<string | null>(null);
+  const [currencySym, setCurrencySym] = useState<string | null>(null);
   const router = useRouter();
-  const { storeData, selectedCurrency } = useStoreContext();
   const { getTokenInformation } = useMyContext();
   const symbolSet = useRef(false);
-  useEffect(() => {
-    if (orderId) {
-      const items = orderItems.get(orderId)?.items || null;
-      setActiveCartItems(items);
-    }
-  }, [orderId, orderItems]);
 
-  const noItems =
-    !orderId || !activeCartItems || !Object.keys(activeCartItems).length;
+  useEffect(() => {
+    (async () => {
```

Comment:
Does the following work?
```suggestion
  useEffect(async () => {
```
its looks like you have a unnecessarily nested function here